MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/analytics/AnalyticsTracker.java

```diff
@@ -0,0 +1,215 @@
+package com.mapbox.mapboxandroiddemo.analytics;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.IdentifyMessage;
+import com.segment.analytics.messages.ScreenMessage;
+import com.segment.analytics.messages.TrackMessage;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class abstracts various analytics calls to Segment analytics' Java library.
+ */
+
+public class AnalyticsTracker {
+
+  private Context appContext = null;
+
+  private static volatile AnalyticsTracker analyticsInstance;
+  private static volatile Analytics analytics;
+  private static final String CLICKED_ON_NAV_DRAWER_SECTION_EVENT_NAME = "Clicked on nav drawer section";
```

Comment:
Fixed. All `trackEvent()` strings have turned into extracted constants rather than hardcoded strings. For example, in MainActivity, one tracked event is now `analytics.trackEvent(CLICKED_ON_INFO_MENU_ITEM,
          loggedInOrNot);`

In AnalyticsTracker class:
`  public static final String CLICKED_ON_INFO_MENU_ITEM = "Clicked on info menu item";`